src/validations/rules/Writing_FedRAMP_Validations.md

```diff
@@ -0,0 +1,375 @@
+    </description>
+</role>
+…
+<party uuid="3360e343-9860-4bda-9dfc-ff427c3dfab6" type="person">
+    <name>[SAMPLE]Person Name 1</name>
+    <prop name="job-title" value="Individual's Title"/>
+    <prop name="mail-stop" value="Mailstop A-1"/>
+    <email-address>name@example.com</email-address>
+    <telephone-number>202-000-0001</telephone-number>
+    <location-uuid>27b78960-59ef-4619-82b0-ae20b9c709ac</location-uuid>
+    <member-of-organization>6b286b5d-8f07-4fa7-8847-1dd0d88f73fb</member-of-organization>
+</party>
+…
+<responsible-party role-id="system-owner">
+    <party-uuid>3360e343-9860-4bda-9dfc-ff427c3dfab6</party-uuid>
+    <remarks>
+        <p>Exactly one</p>
+    </remarks>
+</responsible-party>
+```
+and the code block
+```xml
+<sch:rule
+    context="oscal:responsible-party"
+    doc:guide-reference="Guide to OSCAL-based FedRAMP System Security Plans §4.6-§4.11">
+    <sch:assert
+        diagnostics="responsible-party-has-role-diagnostic"
+        doc:guide-reference="Guide to OSCAL-based FedRAMP System Security Plans §4.6-§4.11"
+        id="responsible-party-has-role"
+        role="error"
+        test="exists(//oscal:role[@id eq current()/@role-id])">The role for a responsible party must exist.</sch:assert>
+    <sch:assert
+        diagnostics="responsible-party-has-party-uuid-diagnostic"
+        id="responsible-party-has-party-uuid"
+        role="error"
+        test="exists(oscal:party-uuid)">One or more parties must be identified for a responsibility.</sch:assert>
+    <sch:assert
+        diagnostics="responsible-party-has-definition-diagnostic"
+        id="responsible-party-has-definition"
+        role="error"
+        test="
+        every $p in oscal:party-uuid
+        satisfies exists(//oscal:party[@uuid eq $p])">Every responsible party must be defined.</sch:assert>
+    <sch:assert
+        diagnostics="responsible-party-is-person-diagnostic"
+        doc:guide-reference="Guide to OSCAL-based FedRAMP System Security Plans §4.6-§4.11"
+        id="responsible-party-is-person"
+        role="error"
+        test="
+        if (
+        current()/@role-id = (
+        'system-owner',
+        'authorizing-official',
+        'system-poc-management',
+        'system-poc-technical',
+        'system-poc-other',
+        'information-system-security-officer',
+        'authorizing-official-poc'
+        )
+        )
+        then
+        every $p in oscal:party-uuid
+        satisfies
+        exists(//oscal:party[@uuid eq $p and @type eq 'person'])
+        else
+        true()
+        ">For some roles responsible parties must be persons.</sch:assert>
+</sch:rule>
+```
+The `<rule>` sets the context for the subordinate assertions.
+
+The first (`id="responsible-party-has-role"`) assertion is an existential test for a `<role>` for the `@role-id`. The use of the `//` prefix in the path `//oscal:role[@id eq current()/@role-id]` means "any `<role>` element within the document" matching the context `@role-id`.
```

Comment:
Same comment as above "existential" vs. "logic"

Reply:
Altered a bit to explain the specific test for existence.